### PR TITLE
[FIX] account_edi: search partner based on VAT with spaces

### PR DESCRIPTION
**Pull request by william-andre**
This solution is not perfect. In order to be, we would need to have a
sanitized function that is used to store the number in the database, and
to use the same function to search in it. This will most likely be done
in master with a refactoring of `base_vat`, on which `account` will
depend one way or another.

In the meantime, we need to support cases that were working before these
fixes:
odoo@bfb2436
odoo@e24c5ba

Since these, it was indeed impossible to detect a partner based on his
VAT for Swiss partners if `base_vat` was installed, which is the
default.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_edi/models/account_edi_format.py [hunk 1]
```diff
@@ -512,33 +513,28 @@ def search_with_vat(extra_domain):
...
             # Try to remove the country code prefix from the vat.
-            if not partner and len(normalized_vat) > 2 and normalized_vat[:2].isalpha():
-                country_prefix = normalized_vat[:2]
-                normalized_vat = normalized_vat[2:]
+            if not partner and country_prefix:
                 partner = self.env['res.partner'].search(extra_domain + [
-                    ('vat', '=', normalized_vat),
-                    ('country_id.code', '=', country_prefix.lower()),
+                    ('vat', 'in', (normalized_vat[2:], vat[2:])),
+                    ('country_id.code', '=', country_prefix.upper()),
```
Comment (william-andre) [inline addons/account_edi/models/account_edi_format.py:524]: Maybe I should put this in a separate commit

File: addons/account_edi/models/account_edi_format.py [hunk 2]
```diff
@@ -512,33 +513,28 @@ def search_with_vat(extra_domain):
...
                     ], limit=1)
 
             # The vat could be a string of alphanumeric values without country code but with missing zeros at the
             # beginning.
             if not partner:
-                try:
-                    vat_only_numeric = str(int(normalized_vat))
-                except ValueError:
-                    vat_only_numeric = None
+                vat_only_numeric = int(re.sub('\D', '', vat) or 0)
```
Comment (william-andre) [inline addons/account_edi/models/account_edi_format.py:537]: Note that the regex search below is bypassing the index https://github.com/odoo/odoo/blob/d6eb4752d7fbd22dd9ad744db51444e1392fd1bb/odoo/addons/base/models/res_partner.py#L173

This might be really bad for big databases with millions of partners.

Comment (oco-odoo) [inline addons/account_edi/models/account_edi_format.py:537]: Don't we risk wrong matches because of that? Removing all the non-digits characters might remove more than just formatting characters added by the user for readability.

**Review by oco-odoo — CHANGES_REQUESTED** [review]

**Review by oco-odoo — APPROVED** [review]
@robodoo r+
